### PR TITLE
Add generic Webhook (HTTP POST callback) as notifier

### PR DESCRIPTION
**Pull request by thomae**
This pull request adds a generic Webhook (HTTP POST callback) notifier. This addresses issue #24.

If a secret is specified in the configuration file the current timestamp together with the request body are used to calculate a HMAC-SHA256 signature. The timestamp and signature are then sent as HTTP header so that the receiver can verify that the request is legitimate and the timestamp is not too old (to prevent replay attacks). I have included a receiver example.

### CODE REVIEW

File: nanny.toml [hunk 1]
```diff
@@ -35,6 +35,13 @@ to=""
 enabled=false
 webhookURL=""
 
+[webhook]
+enabled=false
+webhookURL="https://webhook.somewhere.com/webhook"
+webhookURL_all_clear="https://webhook.somewhere.com/webhook"
```
Comment (lunemec) [inline nanny.toml:41]: Please keep these config names consistent `snake_case`, https://github.com/lunemec/nanny/pull/38/files#diff-ab967ab1a2f3a1b769106eeb7bfe892ef0e81d1d27811fa15be08e6749feee1fR80-R86

File: pkg/notifier/webhook.go [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+	if w.WebhookSecret != "" {
+		timestamp := strconv.FormatInt(time.Now().Unix(), 10)
+		payload := append([]byte(timestamp), postBody...)
+		signature := ComputeHmacSha256(w.WebhookSecret, payload)
+
+		request.Header.Set("X-Timestamp", timestamp)
+		request.Header.Set("X-HMAC-SHA256", signature)
+	}
+
+	client := &http.Client{Timeout: time.Second * 10}
```
Comment (lunemec) [inline pkg/notifier/webhook.go:71]: The http client should be created once and re-used, there are 2 options.
Pass created http client from root.go into `func NewWebhook`, or create it in `func NewWebhook` based on config.

This is required to allow the http client to re-use TCP connections and connection pooling. https://golang.org/pkg/net/http/#Client

Comment (lunemec) [inline pkg/notifier/webhook.go:71]: The webhook http client timeout should be configurable.

File: receiver_examples/webhook_receiver_example.go [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+)
+
+func ComputeHmacSha256(secret string, message []byte) string {
+	key := []byte(secret)
+	h := hmac.New(sha256.New, key)
+	h.Write(message)
+	return hex.EncodeToString(h.Sum(nil))
+}
+
+func main() {
```
Comment (lunemec) [inline receiver_examples/webhook_receiver_example.go:22]: Nice to have example!

**Review by lunemec — CHANGES_REQUESTED** [review]
LGTM, just the config thing and we are good to go! Thanks @thomae

File: nanny.toml (hunk 1 above)
Comment (thomae) [reply nanny.toml]: I have changed the configuration names to snake_case.

File: pkg/notifier/webhook.go (hunk 2 above)
Comment (thomae) [reply pkg/notifier/webhook.go]: The http.Client is now created in `func NewWebhook` based on the config.

Comment (thomae) [reply pkg/notifier/webhook.go]: I have added this option to the config file with a default of 10 seconds.

File: nanny.toml [hunk 4]
```diff
@@ -35,6 +35,14 @@ to=""
 enabled=false
 webhookURL=""
 
+[webhook]
+enabled=false
+webhook_url="https://webhook.somewhere.com/webhook"
+webhook_url_all_clear="https://webhook.somewhere.com/webhook"
+webhook_secret=""
+request_timeout=10
```
Comment (lunemec) [inline nanny.toml:43]: You can use "10s" and have this be `time.Duration` so that you know what is the unit of this.

Comment (thomae) [reply nanny.toml]: I have now included the unit. 🙂

**Comment by lunemec** [comment]
@thomae awesome work. Can you try to run it? Does the `10s` work like this, for some reason I think it has to be inside double-quotes: `"10s"`, but I'm not certain if this works too.

**Review by lunemec — APPROVED** [review]
LGTM

**Comment by thomae** [comment]
I have now changed it to "10s". Both variants worked for me but GitHub highlighted the `s` in red. Probably because it wouldn't be valid in a TOML-file.

**Comment by PhilipSchmid** [comment]
@thomae Do you perhaps also want to bump the [version](https://github.com/lunemec/nanny/blob/master/VERSION)? Besides that, LGTM.

Thanks! Very appreciated feature!

**Comment by PhilipSchmid** [comment]
Awesome, thanks @thomae.

**Pull request merged**